zrml/prediction-markets/src/lib.rs

```diff
@@ -1281,9 +1315,16 @@ mod pallet {
                     )?;
                 }
 
+                // TODO(#372): Allow court with global disputes.
                 // ensure, that global disputes controls the resolution now
                 // it does not end after the dispute period now, but after the global dispute end
-                Self::remove_last_dispute_from_market_ids_per_dispute_block(&disputes, &market_id)?;
+                if let Some(auto_resolve_block) =
+                    T::SimpleDisputes::get_auto_resolve(&disputes, &market_id, &market)?
+                {
+                    MarketIdsPerDisputeBlock::<T>::mutate(auto_resolve_block, |ids| {
+                        remove_item::<MarketIdOf<T>, _>(ids, &market_id);
+                    });
+                }
```

Comment:
I see your point. You like to strongly couple simple-disputes with global disputes. Yes indeed, this is a code smell, but this PR is already far too complex, I guess.

Reply:
https://github.com/zeitgeistpm/zeitgeist/issues/906

Reply:
The purpose of this code block is to delete the normal simple disputes resolution time in order to put in the global dispute resolution time for this market. So in a nutshell: `on_resolution` is called not after one day, but three days. This allows the Zeitgeist holders to participate in a global dispute.

Reply:
When I want to delete the auto resolution for simple-disputes, I need to provide another API call just for the exact same functionality, so that every MDM has to implement it. But I can also do it the way above, which does not require such function.

Reply:
Global disputes is kind of a special feature which gets tested with simple-disputes on battery station. Because I assume it will never be implemented like that on main-net, I think it's okay to have it like that right now. I expect, that the `start_global_disputes` functionality changes in the near future with `court` a lot. So maybe, it's not such a big deal to add a TODO here.

Reply:
cb038d8a97aaa2e9cc58e6747cfc16a3e2590b75